src/index.ts

```diff
@@ -116,7 +120,14 @@ function resolveWebDependency(dep: string, isExplicit: boolean, isOptimized: boo
     );
   }
   if (dependencyStats.isFile()) {
-    return nodeModulesLoc;
+    const locations: DependencyLoc = {};
+    const isJSFile = ['.js', '.mjs', '.cjs'].includes(path.extname(dep));
+    if (isJSFile) {
+      locations.depLoc = dep;
```

Comment:
Keep current behavior, where `depLoc` is returned as the fully resolved, absolute path `nodeModulesLoc`.

Reply:
Will push this with the next commit.